### PR TITLE
fix: fix tab styling on resources page

### PR DESCRIPTION
**Pull request by WillKocur**
The end tags were missing for a few elements in the browse template. I think the browser was autoclosing these tags, but it meant that the browse container wasn't closing until near the end of the page. 

For homepage tabs, this PR:
- Adds in the end tags
- Fixes the resulting broken styling (unwanted border-bottom on browse container)
- Removes unnecessary container
- Use new list mixins
- Use each loop for tabs

For resources:
- Applies this new styling to the resources page
- Adjust resources page styling

### CODE REVIEW

File: app/assets/stylesheets/components/browse.scss [hunk 1]
```diff
@@ -8,18 +8,28 @@
...
+    @include list-item--row(0);
+    // For browse__tab:hover to inherit and overwrite default hover styling
+    color: $white;
+
+    display: inline-block;
   }
 
   @include e(tab) {
+    @include link-reset;
     appearance: none;
```
Comment (nym1k) [inline app/assets/stylesheets/components/browse.scss:33]: Perhaps we could update the button-reset mixin and use it here instead? or combine the two somehow

**Review by nym1k — CHANGES_REQUESTED** [review]
Looks good, just one suggestion but happy to keep as is

File: app/assets/stylesheets/components/browse.scss (hunk 1 above)
Comment (WillKocur) [reply app/assets/stylesheets/components/browse.scss:33]: I did wonder about that. I chose to keep this separate as the color inherit styling could lead to unwanted behaviour, so wouldn't want to apply that to lots of buttons. If that makes sense?

Comment (nym1k) [reply app/assets/stylesheets/components/browse.scss:33]: In theory, we shouldn't be using the default button mixin, and only ever the specific button we need (which is what applies the colour), so I think we can get away with using a color inherit/currentColor property here

Comment (nym1k) [reply app/assets/stylesheets/components/browse.scss:33]: We could also have link-reset and button-reset, and use link-reset within button-reset?

Comment (WillKocur) [reply app/assets/stylesheets/components/browse.scss:33]: The only thing is that it inherits from the parent. So for example:

```
<div style="color: black">
  <button class="button">button</button>
</div>


.button {
  @include link-reset;
  color: red;
}
```

In this scenario on hover the button text will go black, but the desired effect is that it remains red. So to achieve this the button would need a wrapper with color: red. So it's quite hacky and could lead to buttons actually getting hover colors that differ from the normal color of the button in a non-consistent way.

Comment (nym1k) [reply app/assets/stylesheets/components/browse.scss:33]: Okay, let's leave as is for now

**Review by nym1k — APPROVED** [review]

**Pull request merged**